Branch 2 calculator v2

yuqiaoluolong opened the pull request:
Branch 2 calculator v2

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp

* 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp:

0xEljh committed:
add DatedFoodEntry

FoodEntry child class that supports datetime functionality.

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp

* 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp: (46 commits)
  Remove Duke.java
  Rename duke package to dietbook
  change FilePathing to ClassPathing
  fixed calculator
  data printing
  Add new line at the end of the file
  Add ending divider
  Update Expected.TXT with welcome message
  Remove I/O testing
  Change main class in build.gradle
  added back duke
  removed index subtraction
  shortened delete case
  checkstyle fix
  v1.0
  no message
  replace constructor java doc with class java doc
  swap order of assertEquals
  add documentation for food class
  pass gradle checkstyle test
  ...

Commit:
refactor location

Commit:
Recommendation of daily calorie intake calculation is added

HengFuYuen commented on src/main/java/seedu/dietbook/calculator/Calculator.java:
```diff
@@ -0,0 +1,129 @@
+                activityScore = 1.27;
+            }
+            break;
+        case HIGH:
```
Perhaps you would like to handle cases where the activity level is high or extreme and code an assertion for the default case.

yuqiaoluolong replied:
Ok, I added the case where the activity level is high. I did not do so cause the paper only give suggestions for 4 activity levels.

HengFuYuen commented on src/main/java/seedu/dietbook/calculator/Calculator.java:
```diff
@@ -0,0 +1,129 @@
+            requirement = 354 - 6.91 * person.getAge() + 9.36 * activityScore * person.getOriginalWeight()
+                    + 726 * person.getHeight() / 100;
+            break;
+        default:
```
Perhaps you would like to use an assertion for the default case.

yuqiaoluolong replied:
Ok, thanks for reminder.

HengFuYuen commented on src/main/java/seedu/dietbook/calculator/Calculator.java:
```diff
@@ -0,0 +1,129 @@
+
+        switch (person.getGender()) {
+        case MALE:
+            requirement = 662 - 9.53 * person.getAge() + 15.91 * activityScore * person.getOriginalWeight()
```
Maybe you can consider using current weight instead of original weight for the calculations.

HengFuYuen reviewed (commented):
Perhaps some changes would make the calculations more accurate.

0xEljh committed:
add date filtering

- FoodList returns List for public methods. Tweak to calculator to accept List instead of ArrayList made.
- date filtering functions added (to filter between a range or just after a date)
- List function class added to contain constantly reused FP methods on lists.

0xEljh committed:
update FoodListTest time test

Some computers execute too quickly such that there is not neglible difference in time elapsed. Added a short 1 second sleep to prevent this from causing errors.

0xEljh committed:
update FoodListTest to run slower

Executing too quickly affects use of LocalDateTime.now() in tests.

0xEljh committed:
Merge branch 'master' into FoodList-Date-Branch

0xEljh committed:
Resolve merge conflicts

Commit:
updated calculation of recommended daily calorie intake.
assertion about the empty name.

Commit:
added the methods to calculate food nutrients from food added within a certain time period (2 methods).

Commit:
increased accuracy of the calculation by editing the case handling of activityScore.
added 2 assertion statements for the initialization of activityScore and requirement.

Commit:
changed from using original weight to calculate recommended daily calorie intake.

Commit:
automatically changes.

Commit:
Merge branch 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp into branch-2-calculatorv2

* 'master' of https://github.com/AY2021S1-CS2113-T14-4/tp: (76 commits)
  Resolve merge conflicts
  Shift readCommand method over to Ui
  Update quick start and features section
  Update faq and command summary section
  checkstyle
  redundant import
  Bug Fixes and new sub-features
  Update user guide
  Revert "Fix bug for checking of empty inputs"
  Fix bug for checking of empty inputs
  Update parser to support others gender
  Update parser to add current weight parameter
  Change word
  Reorder the outputs
  Fix checkstyle error
  Add methods to calculate nutrient given start date
  Remove bug caused by wrong input order
  Shorten JavaDocs for several methods
  Update printEditedPersonInfo method
  Add printWelcomeBackMessage method
  ...

# Conflicts:
#	src/main/java/seedu/calculator/Calculator.java
#	src/main/java/seedu/dietbook/Parser.java

Commit:
deleted classes not included in the current main from tp_repo

Commit:
Merge branch 'FoodList-Date-Branch' of https://github.com/AY2021S1-CS2113-T14-4/tp into branch-2-calculatorv2

* 'FoodList-Date-Branch' of https://github.com/AY2021S1-CS2113-T14-4/tp:
  Resolve merge conflicts
  update FoodListTest to run slower
  update FoodListTest time test
  add date filtering
  add DatedFoodEntry

# Conflicts:
#	src/main/java/seedu/calculator/Calculator.java

Commit:
resolve static problems

Commit:
updated with new person format

Commit:
resolved name violation

HengFuYuen reviewed (approved):
Current commit LGTM.

0xEljh commented on src/main/java/seedu/dietbook/list/FoodList.java:
```diff
@@ -136,7 +137,7 @@ public boolean clear() {
      * Obtain list of foods consumed within the range of a specified timing.
      */
-    public List<Food> getFoodsInDateTimeRange(LocalDateTime start, LocalDateTime end) {
+    public static List<Food> getFoodsInDateTimeRange(LocalDateTime start, LocalDateTime end) {
```
I disagree with changing this to a static method. Instantiate the FoodList object instead (or get a pointer to the object via argument parameters) to call this method.

0xEljh commented on src/test/java/seedu/dietbook/calculator/CalculatorTest.java:
```diff
@@ -50,4 +53,13 @@ void calculateFat_foodListOfThreeItems_sumOfFat() {
         Calculator calculator = new Calculator(foodList);
         assertEquals(0, calculator.calculateFat());
     }
```
Consider writing tests for the newly added DateTime functions.

0xEljh commented on src/main/java/seedu/dietbook/calculator/Calculator.java:
```diff
@@ -0,0 +1,297 @@
+package seedu.dietbook.calculator;
```
Recommend splitting the class/avoiding a do-it-all class if you find this applicable.

0xEljh reviewed (commented):
Overall do not recommend branch merge as-is.
- More tests recommended.
- Conversion of FoodList to static should either be reverted or made consistent.
